src/_pytest/_code/code.py

```diff
@@ -589,6 +589,31 @@ def exconly(self, tryshort: bool = False) -> str:
         representation is returned (so 'AssertionError: ' is removed from
         the beginning).
         """
+
+        def _get_single_subexc(
+            eg: BaseExceptionGroup[BaseException],
+        ) -> BaseException | None:
+            res: BaseException | None = None
+            for subexc in eg.exceptions:
+                if res is not None:
+                    return None
+
+                if isinstance(subexc, BaseExceptionGroup):
+                    res = _get_single_subexc(subexc)
+                    if res is None:
+                        # there were multiple exceptions in the subgroup
+                        return None
+                else:
+                    res = subexc
+            return res
+
+        if (
+            tryshort
+            and isinstance(self.value, BaseExceptionGroup)
+            and (subexc := _get_single_subexc(self.value)) is not None
+        ):
+            return f"[in {type(self.value).__name__}] {subexc!r}"
```

Comment:
```suggestion
            return f"{subexc!r} [single exception in {type(self.value).__name__}]"
```
as discussed in the issue

Reply:
fixed, but noted my disagreement in the issue :)